HW04/Pavel_Mishchenko/spec/mentor_spec.rb

```diff
@@ -0,0 +1,210 @@
+  describe '#notify_student' do
+    context "when method hasn't arguments" do
+      subject { mentor.notify_student }
+
+      it 'raises ArgumentError' do
+        expect { subject }.to raise_error(ArgumentError)
+      end
+    end
+
+    context 'when mentor notifies student' do
+      subject do
+        mentor.notify_student(student, 'TestDescription')
+        student.notifications_list
+      end
+
+      it 'changes notification list of student' do
+        expect(subject).to include(an_object_having_attributes(description: 'TestDescription'))
+      end
+    end
+  end
+
+  describe '#add_homework' do
+    subject { mentor.add_homework(title: 'HW04', description: 'Task#1', student: student) }
+
+    context 'when mentor adds homework' do
+      it 'returns a new homework' do
+        expect(subject).to be_kind_of(Homework)
+      end
+
+      it 'creates new notification to student' do
+        expect { subject }.to change(student, :notifications_list)
+      end
+    end
+
+    context "when method hasn't arguments" do
+      subject { mentor.add_homework }
+
+      it 'raises ArgumentError' do
+        expect { subject }.to raise_error(ArgumentError)
+      end
+    end
+
+    context 'when student is not Student.class instance' do
+      let(:student) { 404 }
+
+      it 'raises NoMethodError' do
+        expect { subject }.to raise_error(NoMethodError)
+      end
+    end
+  end
+
+  describe '#mark_as_read!' do
+    context 'when notifiications is unreaded' do
+      before do
+        mentor.notifications_list.push(notification1)
+        mentor.notifications_list.push(notification2)
+      end
+
+      it 'makes all notifictions as readed' do
+        mentor.mark_as_read!
+        expect(mentor.notifications_list).to all(have_attributes(status: 'read'))
+      end
+    end
+
+    context 'when notifiications is read' do
+      subject { mentor.mark_as_read! }
+
+      before do
+        notification1.status, notification2.status = 'read'
+        mentor.notifications_list.push(notification1)
+        mentor.notifications_list.push(notification2)
+      end
+
+      it "doesn't make changes" do
+        expect { subject }.not_to change(mentor, :notifications_list)
+      end
+    end
+  end
+
+  describe '#reject_to_work!' do
+    subject { mentor.reject_to_work!(homework) }
+
+    let(:homework) do
+      Homework.new(title: 'TestTitle',
+                   description: 'TestDescription',
+                   student: student,
+                   mentor: mentor)
+    end
+
+    context "when method hasn't arguments" do
+      subject { mentor.reject_to_work! }
+
+      it 'raises ArgumentError' do
+        expect { subject }.to raise_error(ArgumentError)
+      end
+    end
+
+    context 'when homework is not Homework.class instance' do
+      let(:homework) { 404 }
+
+      it 'raises NoMethodError' do
+        expect { subject }.to raise_error(NoMethodError)
+      end
+    end
+
+    context 'when mentor reject homework' do
```

Comment:
the same thing: try use the opposite contexts
for instance, `when arguments are valid`